File: GO/Estufa.go
package main

import "fmt"

type Estufa struct {
	Electrodomestico // Aca se hace la "herencia"(Es una composicion)
	Peso             float64
	Temperatura      float64
	Sirve            bool
}

// Constructor (Go no tiene constructores automáticos)
func NewEstufa(peso float64, temperatura float64, sirve bool) *Estufa {
	return &Estufa{
		Peso:        peso,
		Temperatura: temperatura,
		Sirve:       sirve,
	}
}
func (e *Estufa) Prevencion(temp float64) {
	if !e.Sirve {
		fmt.Println("No sirve mijo")
	} else if temp > 200 {
		e.Encender(false)
	} else {
		e.Encender(true)
	}
}

// Sobrescribimos Encender (polimorfismo)
func (e *Estufa) Encender(encendido bool) {
	e.encendido = encendido
	if encendido {
		fmt.Println("Se enciende")
	} else {
		fmt.Println("Se apaga")
	}
}

// Implementa el otro método abstracto
func (e Estufa) Funcionar() {
	fmt.Println("Funciono")
}

File: Java/poocorredor/Estufa.java
    /*
    * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
    * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
    */
    package poocorredor;

    /**
     *
     * @author LENOVO
     */
    public class Estufa extends Electrodomestico {
        private double peso;
        private double temperatura;
        private boolean sirve;
        
        public Estufa(double peso, double temperatura, boolean sirve){
            this.peso = peso;
            this.temperatura = temperatura;
            this.sirve = sirve;
        }
        
        public void prevencion(double temperatura){
            if (sirve == false) {
                System.out.println("No sirve mijo");
            }else if (temperatura>200) {
                encender(false);
            }else{
                encender(true);
            }
        }
        @Override
        public void encender(boolean encendido) {
            if (encendido==true) {
                System.out.println("Se enciende");
            }else{
                System.out.println("se apaga");
            }
        }
        @Override
        public void funcionar() {
            System.out.println("Funciono");
            
        }
    }
